src/domains.ts

```diff
@@ -0,0 +1,319 @@
+import Errors from './errors'
+
+export interface Domain {
+  name?: string
+  id?: string
+  alias?: string
+  email?: string
+}
+
+interface QueryParams {
+  offset: number
+  limit: number
+}
+
+interface DomainsInterface {
+  domains: Domain[]
+  page: PageRes
+}
+
+interface PageRes {
+  total: number
+  offset: number
+  limit: number
+}
+
+export default class Domains {
+  // Domains API client
+
+  private readonly domainsUrl: URL
+  private readonly contentType: string
+  private readonly domainsEndpoint: string
+  private readonly domainError: Errors
+
+  public constructor (domainsUrl: string) {
+    this.domainsUrl = new URL(domainsUrl)
+    this.contentType = 'application/json'
+    this.domainsEndpoint = 'domains'
+    this.domainError = new Errors()
+  }
+
+  public async CreateDomain (domain: Domain, token: string): Promise<Domain | undefined> {
```

Comment:
we should not provision for a return of `undefined` assert the type to be `Domain` only. We should only either return `domain` or an `error`, nothing else.

Apply this to the others as well